src/System.IO.Compression/src/System/IO/Compression/DeflateZLib/Deflater.cs

```diff
@@ -16,7 +16,7 @@ namespace System.IO.Compression
     /// </summary>
     internal sealed class Deflater : IDisposable
     {
-        private ZLibNative.ZLibStreamHandle _zlibStream;
+        private ZLibNative.ZLibStreamHandle _zlibStream = null!;
```

Comment:
Please add a comment here that this is initialized by a method that's called from the ctor